### PR TITLE
Update weights for Statemine, Statemint and  Westmint

### PR DESCRIPTION
**Pull request by NachoPal**
Related to: #626 
 
Update weights for Statemine, Statemint and  Westmint

Also `pallet_uniques` benchmarks added to Statemint and Westmint

### CODE REVIEW

File: polkadot-parachains/statemine/src/weights/pallet_utility.rs [hunk 1]
```diff
@@ -29,16 +46,16 @@ use sp_std::marker::PhantomData;
 pub struct WeightInfo<T>(PhantomData<T>);
 impl<T: frame_system::Config> pallet_utility::WeightInfo for WeightInfo<T> {
 	fn batch(c: u32, ) -> Weight {
-		(16_177_000 as Weight)
-			// Standard Error: 0
-			.saturating_add((4_582_000 as Weight).saturating_mul(c as Weight))
+		(31_019_000 as Weight)
+			// Standard Error: 3_000
+			.saturating_add((7_923_000 as Weight).saturating_mul(c as Weight))
```
Comment (apopiak) [inline polkadot-parachains/statemine/src/weights/pallet_utility.rs:51]: this seems quite a bit heavier

Comment (apopiak) [reply polkadot-parachains/statemine/src/weights/pallet_utility.rs:51]: might be the events from here? https://github.com/paritytech/substrate/pull/9462

**Review by apopiak — COMMENTED** [review]
found some weight changes, some of them I have found a plausible source for

File: polkadot-parachains/statemine/src/weights/pallet_collator_selection.rs [hunk 2]
```diff
@@ -28,48 +45,66 @@ use sp_std::marker::PhantomData;
...
+	// Storage: CollatorSelection LastAuthoredBlock (r:0 w:1)
 	fn register_as_candidate(c: u32, ) -> Weight {
-		(72_345_000 as Weight)
-			// Standard Error: 0
-			.saturating_add((197_000 as Weight).saturating_mul(c as Weight))
-			.saturating_add(T::DbWeight::get().reads(4 as Weight))
+		(76_016_000 as Weight)
+			// Standard Error: 1_000
+			.saturating_add((151_000 as Weight).saturating_mul(c as Weight))
+			.saturating_add(T::DbWeight::get().reads(5 as Weight))
```
[the remark this answers is not in the document]
Comment (NachoPal) [reply polkadot-parachains/statemine/src/weights/pallet_collator_selection.rs:75]: In this PR: #498 An extra read was introduced here: https://github.com/paritytech/cumulus/pull/498/files#diff-a0359a0add814d66aac74493f7e17aefdcee21d3de1ae592e97059082b6feedfR337

File: polkadot-parachains/statemine/src/weights/pallet_collator_selection.rs [hunk 3]
```diff
@@ -28,48 +45,66 @@ use sp_std::marker::PhantomData;
...
 	}
+	// Storage: CollatorSelection Candidates (r:1 w:1)
+	// Storage: CollatorSelection LastAuthoredBlock (r:0 w:1)
 	fn leave_intent(c: u32, ) -> Weight {
-		(55_446_000 as Weight)
-			// Standard Error: 0
-			.saturating_add((153_000 as Weight).saturating_mul(c as Weight))
+		(76_011_000 as Weight)
+			// Standard Error: 1_000
+			.saturating_add((202_000 as Weight).saturating_mul(c as Weight))
```
[the remark this answers is not in the document]
Comment (NachoPal) [reply polkadot-parachains/statemine/src/weights/pallet_collator_selection.rs:83]: Same PR: #498
https://github.com/paritytech/cumulus/pull/498/files#diff-a0359a0add814d66aac74493f7e17aefdcee21d3de1ae592e97059082b6feedfR362

File: polkadot-parachains/statemine/src/weights/pallet_collator_selection.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (apopiak) [reply polkadot-parachains/statemine/src/weights/pallet_collator_selection.rs:75]: so `T::MinCandidates` is a storage item and induces a read?

[the remark this answers is not in the document]
Comment (NachoPal) [reply polkadot-parachains/statemine/src/weights/pallet_collator_selection.rs:75]: I think you are looking at the wrong line, it is `337`, and the extra read comes from `T::ValidatorRegistration::is_registered(&validator_key)` that will take a look at `pallet_session` storage looking for the keys.

[the remark this answers is not in the document]
Comment (apopiak) [reply polkadot-parachains/statemine/src/weights/pallet_collator_selection.rs:75]: ah, oops, :ok:

File: polkadot-parachains/statemine/src/weights/pallet_collator_selection.rs [hunk 4]
```diff
@@ -28,48 +45,66 @@ use sp_std::marker::PhantomData;
...
-			.saturating_add(T::DbWeight::get().reads((1 as Weight).saturating_mul(r as Weight)))
+			// Standard Error: 4_417_000
+			.saturating_add((24_857_000 as Weight).saturating_mul(r as Weight))
+			// Standard Error: 4_417_000
+			.saturating_add((115_986_000 as Weight).saturating_mul(c as Weight))
 			.saturating_add(T::DbWeight::get().reads((2 as Weight).saturating_mul(c as Weight)))
-			.saturating_add(T::DbWeight::get().writes((2 as Weight).saturating_mul(r as Weight)))
-			.saturating_add(T::DbWeight::get().writes((2 as Weight).saturating_mul(c as Weight)))
+			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(r as Weight)))
+			.saturating_add(T::DbWeight::get().writes((1 as Weight).saturating_mul(c as Weight)))
```
[the remark this answers is not in the document]
Comment (apopiak) [reply polkadot-parachains/statemine/src/weights/pallet_collator_selection.rs:108]: ah, you changed that benchmark in https://github.com/paritytech/cumulus/pull/639

**Review by apopiak — APPROVED** [review]
All changes seem accounted for

**Pull request merged**